src/NuGet.Clients/PackageManagement.VisualStudio/ProjectSystems/VSMSBuildNuGetProjectSystem.cs

```diff
@@ -889,7 +884,14 @@ public Task ExecuteScriptAsync(PackageIdentity identity, string packageInstallPa
 
         public virtual void BeginProcessing()
         {
-            ProjectBuildSystem?.StartBatchEdit();
+            if (ProjectBuildSystem != null)
+            {
+                NuGetUIThreadHelper.JoinableTaskFactory.Run(async delegate
+                {
+                    await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+                    ProjectBuildSystem.StartBatchEdit();
```

Comment:
Is `ProjectBuildSystem` guaranteed to not become null in between your check and switching to the UI thread? If it could become null, you should also check after you get to the UI thread, just before dereferencing it to avoid throwing NRE.